### ISSUE TITLE
Use of "gamelist-only" should disable "file exists" checks on startup

### ISSUE THREAD

**Issue by Enverex** [Aloshi/EmulationStation#401]
I took it for granted that this was how it worked by default but apparently not.

When using "gamelist-only", ES should not bother scanning for files itself, it should assume that the gamelist is correct (hence telling it just to use the gamelist and not scan). The rom folder and file extension options in the es_systems file shouldn't matter when using this flag as they won't be used.

This would also solve issue https://github.com/Aloshi/EmulationStation/issues/370 as it wouldn't matter what the ROM file path was set to, it would just "run" it without checking.

**Comment by MongoBongo** [comment]
This sounds like it would be a great thing for large collections, Maybe something like this can be implemented? @Aloshi

**Comment by kyleabaker** [comment]
+1

**Comment by robertybob** [comment]
Is there any progress on this? If it'll speed up the loading of ES for people with 1000+ games then I can't wait !

**Comment by Enverex** [comment]
Any news on this? I fear the great Aloshi has abandoned us.

**Comment by kasketis** [comment]
Any news????

**Comment by Enverex** [comment]
Aloshi isn't working on this project anymore so unfortunately unless someone else picks up the mantle, it's dead in the water.

**Comment by LynoDesu** [comment]
Hi, I've just started using EmulationStation and this is one of my major annoyances. I have a large ROM collection stored on my NAS and it takes an age to load at startup and to scrape.

I'm going to fork this repo and work on a fix to stop it checking for file exists in gamelist-only mode.

**Comment by LynoDesu** [comment]
OK, it seems that this should already be supported, though as a launch parameter not a cfg setting.

It should do this if you launch ES with the argument:  --ignore-gamelist

**Comment by Enverex** [comment]
Surely that's back-to-front, we want to use the gamelist and ignore the scan, hence the flag "--gamelist-only".

**Comment by LynoDesu** [comment]
In GameList.cpp, this method gets called and iterates through all the files to check whether there already in the XML. As you can see, it doesn't do this if IgnoreGameList is set:

```
void updateGamelist(SystemData* system)
{
	//We do this by reading the XML again, adding changes and then writing it back,
	//because there might be information missing in our systemdata which would then miss in the new XML.
	//We have the complete information for every game though, so we can simply remove a game
	//we already have in the system from the XML, and then add it back from its GameData information...

	if(Settings::getInstance()->getBool("IgnoreGamelist"))
		return;
```

**Comment by Enverex** [comment]
Right, but that's because if you tell it to ignore the gamelist, it'll still do a scan, but it won't compare it to the gamelist (as you've told it to ignore it). So in your example, your gamelist will be ignored and it'll end up scanning your entire library of games, surely? (which isn't what we want).

**Comment by LynoDesu** [comment]
Yes, sorry I'd not seen exactly how that variable was being used.

So I've found the check for whether the file exists when it parses the XML.

If I fix this and create a pull request, will it be reviewed or has Aloshi completely abandoned this repo?

**Comment by Enverex** [comment]
He says he doesn't have time to continue development, but I'd imagine a pull request would take seconds at most.

**Comment by LynoDesu** [comment]
Awesome. I'll change the code and test it after work, tonight, and if all goes well I'll submit a pull request.

**Comment by Enverex** [comment]
Did you ever make the changes in your fork for this? I'm not seeing any commits.

**Comment by LynoDesu** [comment]
No....I made the code changes that should (in theory) work, but when I went to compile the source changes I'd made, it turned out to be a bit of a hassle. I'm compiling in windows and you need a handful of dependencies that you either have to build from source manually or find elsewhere. If it's easier on Unix/Linux then I'll build on my Macbook when I have time

**Comment by Enverex** [comment]
If you push that change to your repo, I'll try compiling it and see how it turns out.

**Comment by LynoDesu** [comment]
OK, I've pushed the changes. No idea if it'll work or not, I've a feeling that it might not even load the game list correctly now. But since I have no way or building or debugging (easily) I can't really say
